README.md

```diff
@@ -119,7 +119,32 @@ Sometimes it may happen that you need to process the relationship before renderi
 ```ruby
 class BlogPostSerializer < BaseSerializer
   lazy_has_many :comments do |serializer|
-    serializer.lazy_comments.map(&:decorate)
+    -> { serializer.lazy_comments.map(&:decorate) }
+   end
+end
+```
+
+Keep attention that custom finder should be returned in a form of lambda,
+in order to avoid redundant SQL queries when `include_data` AMS setting appears to be `false`:
+
+```ruby
+class BlogPostSerializer < BaseSerializer
+  lazy_has_many :comments do |serializer|
+    include_data :if_sideloaded
+    -> { serializer.lazy_comments.map(&:decorate) }
+   end
+end
+```
+
+Feel free to skip custom lazy finder for association if your goal is just to
+define `include_data` setting nor some links/metas:
```

Comment:
```suggestion
define `include_data` setting to render some links/metas:
```

?

Reply:
`include_data` is not about links and metas, but mostly about skipping so-called json `relationships`.

By default AMS will add to response the primary keys for each association defined in serializer. I.e.

```ruby
class OrganizationSerializer
  has_many :users
end

OrganizationSerializer.new(org).as_json
# {data: {relationships: { users: { [{id: 1}, {id: 2}, ..., {id: 123_567_89} ] } } } }
```

That happens always even *without* explicit `?include=users`, that produces `SELECT * FROM users` query on the way and memory allocation for `123_567_89` AR objects.

`include_data = false or :if_sideloaded` is a way to survive.

I never tried believe that `include_data = :if_sideloaded` can be combined with links and metas

Reply:
True, you're right, I missed the point :) But TBH here I mostly meant that this sentence is not a correct english sentence and tried to rework it a bit. Would you be able to rephrase it a bit?

Reply:
will try, thanks! )